lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -6,15 +6,16 @@
             'gettext',
             'js/views/fields',
             'teams/js/models/team',
+            'common/js/components/utils/view_utils',
             'text!teams/templates/edit-team.underscore'],
-        function (Backbone, _, gettext, FieldViews, TeamModel, editTeamTemplate) {
+        function (Backbone, _, gettext, FieldViews, TeamModel, ViewUtils, editTeamTemplate) {
             return Backbone.View.extend({
 
                 maxTeamNameLength: 255,
                 maxTeamDescriptionLength: 300,
 
                 events: {
-                    'click .action-primary': 'createOrUpdateTeam',
+                    'click .action-primary': function(event) {this.disableRun(event, this.createOrUpdateTeam);},
```

Comment:
This is much better now. I think you should move the anonymous function into ``disableRun`` and then you can make this line just:

    'click .action-primary': this.disableRun('createOrUpdateTeam')

I believe you have to make it a string as you mentioned above, because the field isn't available at the time of creation. As you suggested, it would be even better to have this be ``ViewUtils.disableRun``. In fact, I'd go with ``ViewUtils.withDisabledButton`` or ``ViewUtils.withDisabledElement`` to make it clearer what it is doing for you.